### PR TITLE
Errors handler

### PR DESCRIPTION
**Pull request by LORD-ME-CODE**
```python
from pyrogram import Client, filters

app = Client(...)
```
## Use with decorator
```python
@app.on_error()  # Global error handler, all other errors will be sent to it
async def err_handler(_, e: Exception, update):
    print(e, 'handled')
    print(update)


@app.on_error(ValueError)  #  Also you can use tuple with many exception classes, example: (ValueError, TypeError)
async def custom_err_handler(_, e: ValueError, update):
    print(e, 'custom handled')
    print(update)
```

## Use without a decorator

```python
value_error_handler = ErrorHandler(custom_err_handler, ValueError)  #  Also you can use tuple with many exception classes, example: (ValueError, TypeError)
global_error_handler = ErrorHandler(err_handler)  # Global error handler, all other errors will be sent to it
app.add_handler(value_error_handler)
app.add_handler(global_error_handler)
```

## Startup example

```python
@app.on_message(filters.command('testerror'))
async def msg_handler(_, m):
    raise Exception('TestError 123')


@app.on_message(filters.command('valueerror'))
async def custom_msg_handler(_, m):
    raise ValueError('TestError 123')

app.run()
```

## Remove error handler

```python
app.remove_error_handler(value_error_handler)
# Or you can use
app.remove_error_handler(error=ValueError)  # to remove all error handlers with ValueError in .errors
# For remove global error handler, you can use:
app.remove_error_handler()
```

### CODE REVIEW

File: pyrogram/handlers/error_handler.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            Pass a function that will be called when a new Exception arrives. It takes *(client, error)*
+            as positional arguments (look at the section below for a detailed description).
+    Other parameters:
+        client (:obj:`~pyrogram.Client`):
+            The Client itself, useful when you want to call other API methods inside the message handler.
+        error (:obj:`Exception`):
+            The received error.
+    """
+
+    def __init__(self, callback: callable, filters=None):
```
[the remark this answers is not in the document]
Comment (AndrielFR) [reply pyrogram/handlers/error_handler.py:40]: wouldn't it be better, `errors=None`? to use it like:
```python
def floodwait_handler(_, e: FloodWait, update):
    print(e.x)

app.add_handler(ErrorHandler(floodwait_handler, FloodWait))
```

File: pyrogram/handlers/error_handler.py [hunk 2]
```diff
@@ -35,5 +33,9 @@ class ErrorHandler(Handler):
             The received error.
     """
 
+    # TODO ⬇️
+    """For a nicer way to register this handler, have a look at the
+    :meth:`~pyrogram.Client.on_error` decorator."""
```
Comment (AndrielFR) [inline pyrogram/handlers/error_handler.py:38]: ok, it's good to have an easiest way to handle errors.

**Review by AndrielFR — CHANGES_REQUESTED** [review]

**Comment by AndrielFR** [comment]
this is a better way to implement it than https://github.com/pyrogram/pyrogram/pull/597

File: pyrogram/handlers/error_handler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LORD-ME-CODE) [reply pyrogram/handlers/error_handler.py]: Added

[the remark this answers is not in the document]
Comment (AndrielFR) [reply pyrogram/handlers/error_handler.py]: nice

File: pyrogram/handlers/error_handler.py (hunk 2 above)
Comment (LORD-ME-CODE) [reply pyrogram/handlers/error_handler.py]: complete

**Comment by Danipulok** [comment]
@delivrance why not merge it?
Seems really good and useful. Can be one of v2.X features

**Comment by aditya-yadav-27** [comment]
Niice work, really liked it. Hope this get merged soon

**Comment by Danipulok** [comment]
@delivrance any update?

**Pull request closed without merging**